src/ngram_matches.cc

```diff
@@ -69,23 +55,20 @@ namespace fuzzy
 
       // Get or create the AgendaItem corresponding to the sentence (of the suffix that matched)
       const auto sentence_id = _suffixArray.suffixid2sentenceid()[i].sentence_id;
-      auto* agendaItem = get_agendaitem(sentence_id);
-      if (!agendaItem)
-        agendaItem = new_agendaitem(sentence_id, _p_length);
+      auto& agendaItem = _psentences.emplace(sentence_id,
+                                             AgendaItem(sentence_id, _p_length)).first.value();
 
       // The match will update the AgendaItem entry only if its length is the longest to date.
-      if (range.match_length <= static_cast<std::size_t>(agendaItem->maxmatch))
```

Comment:
I don't know if this makes a difference but this line prevented a write on the `agendaItem->maxmatch`, maybe we should use `if` instead of `max` at the end.

Reply:
Thanks, it makes sense. I restored the `if` to avoid the max.